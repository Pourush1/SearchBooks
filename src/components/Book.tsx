import React from 'react';

interface IBooksProps {
  title: string;
  authors?: string[];
  publishedDate?: string;
  thumbnail?: string;
  publisher?: string;
  description?: string;
}

const Book = ({
  title,
  authors,
  publishedDate,
  thumbnail,
  publisher,
  description
}: IBooksProps) => {
  return (
    <div className="col md-2">
      <div className="card">
        <p>{title}</p>
        <div className="card-body">
          <h1>{publishedDate}</h1>
          <h1>{publisher}</h1>
        </div>
      </div>
    </div>
  );
};

export default Book;
