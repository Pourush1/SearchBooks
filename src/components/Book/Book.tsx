import React from 'react';
import { IBook } from '../../book-search/BookSearch';
import moment from 'moment';

interface IAddToWish {
  addToWishList: (book: IBook) => void;
  disableAddToWishListButton?: boolean;
  index: number;
}

const Book: React.FC<IBook & IAddToWish> = ({
  title,
  publishedDate,
  thumbnail,
  publisher,
  addToWishList,
  disableAddToWishListButton,
  index,
}) => {
  return (
    <div className="col-sm-3 pb-3" key={index}>
      <div className="card p-3">
        <img
          className="card-img-top mb-3"
          src={thumbnail}
          alt="book cover"
        ></img>
        <div className="card-body">
          <p className="font-weight-bold">{title}</p>
          <p>{moment(publishedDate).format('MMM Do YYYY')}</p>
          <p>{publisher}</p>
          <button
            onClick={() =>
              addToWishList({ title, thumbnail, publisher, publishedDate })
            }
            className="btn btn-primary"
            disabled={disableAddToWishListButton}
          >
            Add to WishList
          </button>
        </div>
      </div>
    </div>
  );
};

export default Book;
