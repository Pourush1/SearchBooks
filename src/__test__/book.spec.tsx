import React from 'react';
import {
  render,
  RenderResult,
  screen,
  fireEvent
} from '@testing-library/react';
import Book from '../components/Book/Book';

let documentBody: RenderResult;

describe('<Book />', () => {
  const mock = jest.fn(() => {});

  const props = {
    title: 'Java',
    addToWishList: mock,
    published: '2016-11-05'
  };

  beforeEach(() => {
    documentBody = render(<Book {...props} />);
  });

  it('shows initial messages', () => {
    const wishlist = screen.getAllByText('Add to WishList');
    expect(wishlist.length).toEqual(1);
  });

  it('should click wihslist button and call the mock', () => {
    fireEvent.click(screen.getByText('Add to WishList'));
    expect(mock).toHaveBeenCalledTimes(1);
    fireEvent.click(screen.getByText('Add to WishList'));
    expect(mock).toHaveBeenCalledTimes(2);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
