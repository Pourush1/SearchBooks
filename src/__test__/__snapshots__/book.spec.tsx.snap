// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="col-sm-3 pb-3"
    >
      <div
        class="card p-3"
      >
        <img
          alt="book cover"
          class="card-img-top mb-3"
        />
        <div
          class="card-body"
        >
          <p
            class="font-weight-bold"
          >
            Java
          </p>
          <p />
          <p />
          <button
            class="btn btn-primary"
          >
            Add to WishList
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
